### PR TITLE
Path.endsNotWith

### PR DESCRIPTION
**Pull request by segunfamisa**
## Implement Path.endsNotWith #190 

I confirm that I have read the [Contributor Agreements v1.0](https://github.com/robstoll/atrium/blob/master/.github/Contributor%20Agreements%20v1.0.txt), agree to be bound on them and confirm that my contribution is compliant.

### CODE REVIEW

File: domain/builders/atrium-domain-builders-jvm/src/main/kotlin/ch/tutteli/atrium/domain/builders/creating/PathAssertionsBuilder.kt [hunk 1]
```diff
@@ -17,6 +17,9 @@ object PathAssertionsBuilder : PathAssertions {
     override fun <T : Path> endsWith(assertionContainer: Expect<T>, expected: Path) =
         pathAssertions.endsWith(assertionContainer, expected)
 
+    override fun <T : Path> endsNotWith(assertionContainer: Expect<T>, expected: Path) =
```
Comment (robstoll) [inline domain/builders/atrium-domain-builders-jvm/src/main/kotlin/ch/tutteli/atrium/domain/builders/creating/PathAssertionsBuilder.kt]: please `inline` the function (could you do the same for endsWith, I apparently did not see it during the review)

File: misc/specs/atrium-specs-jvm/src/main/kotlin/ch/tutteli/atrium/specs/integration/PathAssertionsSpec.kt [hunk 2]
```diff
@@ -112,5 +113,26 @@ abstract class PathAssertionsSpec(
             }
         }
     }
+
+    describeFun(endsNotWith.name) {
```
Comment (robstoll) [inline misc/specs/atrium-specs-jvm/src/main/kotlin/ch/tutteli/atrium/specs/integration/PathAssertionsSpec.kt]: personally I would fuse this describe with the above one as this is the opposite of endsWith. This way you could reuse the `context...` but it's also fine if you leave it this way

**Comment by robstoll** [comment]
@segunfamisa please rebase on master and resolve the conflicts

File: misc/specs/atrium-specs-jvm/src/main/kotlin/ch/tutteli/atrium/specs/integration/PathAssertionsSpec.kt (hunk 2 above)
Comment (robstoll) [reply misc/specs/atrium-specs-jvm/src/main/kotlin/ch/tutteli/atrium/specs/integration/PathAssertionsSpec.kt]: I see that all others are also this way, so let's keep it for now

**Comment by segunfamisa** [comment]
Cool!

Thanks for the review. I'll inline those two functions (completely forgot
it could be used inline as well:)) and rebase.

**Comment by segunfamisa** [comment]
Hey @robstoll 

Sorry, I hastily sent that on the go from my phone. I had thought you meant infix, not inline.

Is there any benefit of inlining it? just asking since I don't see any lambda functions that the compiler could inline. Also I'm getting compiler warnings saying `"...expected performance impact of inlining is insignificant"` 

But I'm also not sure maybe there is something I'm missing.

**Comment by robstoll** [comment]
You should not get a compiler warning as there is `@file:Suppress("OVERRIDE_BY_INLINE", "NOTHING_TO_INLINE")` in the PathAssertionsBuilder file
I inline for backward compatibility reasons, see https://github.com/robstoll/atrium-roadmap/issues/25#issuecomment-526239006 => rough analysis => delegation in case you are curious

**Comment by segunfamisa** [comment]
Thanks for the explanation. I'll have a look at the roadmap link you shared!

**Comment by robstoll** [comment]
@segunfamisa thanks for the contribution :+1:

Fancy another one? (for instance #200 would be a good one to continue with):
https://github.com/robstoll/atrium/issues?q=is%3Aissue+is%3Aopen+label%3Ahacktoberfest+no%3Aassignee

(and don't forget to star Atrium :wink:)

**Pull request merged**